controllers/healthcheck/machine.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthcheck
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate counterfeiter . Client
+type Client interface {
+	Delete(context.Context, runtime.Object, ...client.DeleteOption) error
+}
```

Comment:
Could we use controller runtime fake client instead of creating a new mock client here?

Reply:
Isn't it deprecated?

Reply:
I responded to that here too https://github.com/kubernetes-sigs/cluster-api/pull/2975#issuecomment-622539961, I think it's better to use a narrower interface where possible, and we don't need the integration coverage of a fake object database that the CR client gives us

Reply:
Yes, but we shouldn't worry about it until CR 1.0, and we have https://github.com/kubernetes-sigs/controller-runtime/issues/768 where community is discussing potential approaches